apps/emqx_authz/src/emqx_authz_api_schema.erl

```diff
@@ -16,490 +16,185 @@
-            redis_type => #{type => string,
-                            enum => [<<"cluster">>],
-                            example => <<"cluster">>},
-            pool_size => #{type => integer},
-            auto_reconnect => #{type => boolean, example => true},
-            password => #{type => string},
-            database => #{type => integer},
-            ssl => minirest:ref(<<"ssl">>)
-        }
-    },
-    Mnesia = #{
-        type => object,
-        required => [type, enable],
-        properties => #{
-            type => #{
-                type => string,
-                enum => [<<"redis">>],
-                example => <<"redis">>
-            },
-            enable => #{
-                type => boolean,
-                example => true
-            }
-        }
-    },
-    File = #{
-        type => object,
-        required => [type, enable, rules],
-        properties => #{
-            type => #{
-                type => string,
-                enum => [<<"redis">>],
-                example => <<"redis">>
-            },
-            enable => #{
-                type => boolean,
-                example => true
-            },
-            rules => #{
-                type => array,
-                items => #{
-                  type => string,
-                  example =>
-                      <<"{allow,{username,\"^dashboard?\"},","subscribe,[\"$SYS/#\"]}.\n",
-                        "{allow,{ipaddr,\"127.0.0.1\"},all,[\"$SYS/#\",\"#\"]}.">>
-                }
-            },
-            path => #{
-                type => string,
-                example => <<"/path/to/authorizaiton_rules.conf">>
-            }
-        }
-    },
-    [ #{<<"sources">> => Sources}
-    , #{<<"ssl">> => SSL}
-    , #{<<"http">> => HTTP}
-    , #{<<"built-in-database">> => Mnesia}
-    , #{<<"mongo_single">> => MongoSingle}
-    , #{<<"mongo_rs">> => MongoRs}
-    , #{<<"mongo_sharded">> => MongoSharded}
-    , #{<<"mysql">> => Mysql}
-    , #{<<"postgresql">> => Pgsql}
-    , #{<<"redis_single">> => RedisSingle}
-    , #{<<"redis_sentinel">> => RedisSentinel}
-    , #{<<"redis_cluster">> => RedisCluster}
-    , #{<<"file">> => File}
+-import(hoconsc, [mk/2, ref/1, ref/2, array/1, enum/1]).
+-import(emqx_schema, [mk_duration/2]).
+
+-export([fields/1, authz_sources_types/1]).
+
+fields(http) ->
+    authz_common_fields(http)
+        ++ [ {url, fun url/1}
+           , {method, #{ type => enum([get, post])
+                       , default => get
+                       , converter => fun to_bin/1}}
+           , {headers, fun headers/1}
+           , {body, fun body/1}
+           , {request_timeout, mk_duration("Request timeout", #{default => "30s"})}]
+        ++ maps:to_list(maps:without([ base_url
+                                     , pool_type],
+                                     maps:from_list(emqx_connector_http:fields(config))));
+fields('built-in-database') ->
```

Comment:
maybe a better name? 
```
'built_in_database'
```

Reply:
Maybe change with `password-based` together in another PR.